sbt-scalajs-bundler/src/main/scala/scalajsbundler/util/Commands.scala

```diff
@@ -1,19 +1,36 @@
 package scalajsbundler.util
 
 import sbt.Logger
-import java.io.File
+import java.io.{InputStream, File}
 import scala.sys.process.Process
+import scala.sys.process.BasicIO
 import scala.sys.process.ProcessLogger
 
 object Commands {
 
-  def run(cmd: Seq[String], cwd: File, logger: Logger): Unit = {
+  def run[A](cmd: Seq[String], cwd: File, logger: Logger, outputProcess: InputStream => A): Option[A] = {
+    val toErrorLog = (is: InputStream) => scala.io.Source.fromInputStream(is).getLines.foreach(msg => logger.error(msg))
+
+    // Unfortunately a var is the only way to capture the result
+    var result: Option[A] = None
+    def outputCapture(o: InputStream): Unit = {
+      result = Some(outputProcess(o))
+      ()
+    }
+
     logger.debug(s"Command: ${cmd.mkString(" ")}")
     val process = Process(cmd, cwd)
-    val code = process ! toProcessLogger(logger)
+    val processIO = BasicIO.standard(false).withOutput(outputCapture).withError(toErrorLog)
```

Comment:
Maybe the `run` method of `ProcessBuilder` would be more convenient? http://scala-lang.org/api/current/scala/sys/process/ProcessBuilder.html#run(log:scala.sys.process.ProcessLogger):scala.sys.process.Process

Reply:
i'll check `run`

Reply:
`run` doesn't give you access to the output stream. I think the current version is more usable